internal/cli/deploy.go

```diff
@@ -0,0 +1,215 @@
+	fs.StringVarP(&o.Name, "name", "s", o.Name, "Name of the vm")
+	fs.StringVarP(&o.NetworkName, "network", "n", "default", "Name of the network")
+	fs.StringVarP(&o.PersistenceImagePath, "persistence-image", "p", o.PersistenceImagePath, "Path to the persistence image.If not set it will be generate in tmp folder (size 1G)")
+	fs.StringVarP(&o.QemuUrl, "qemu-url", "p", "qemu:///session", "Url of qemu")
+}
+
+func (o *DeployOptions) Validate(args []string) error {
+	if _, err := uuid.Parse(args[0]); err != nil {
+		return fmt.Errorf("invalid source id: %s", err)
+	}
+
+	if o.Name == "" {
+		// generate a vm like agent-123456
+		o.Name = fmt.Sprintf("agent-%s", uuid.NewString()[:6])
+	}
+
+	if o.PersistenceImagePath == "" {
+		o.PersistenceImagePath = path.Join("/tmp", fmt.Sprintf("persistence-%s.qcow2", uuid.NewString()[:6]))
+	}
+
+	return nil
+}
+
+func (o *DeployOptions) Run(ctx context.Context, args []string) error {
+	if o.ImageFile == "" {
+		tmpFolder, err := os.MkdirTemp("", o.Name)
+		if err != nil {
+			return fmt.Errorf("failed to create temporary folder for image iso: %v", err)
+		}
+		o.ImageFile = path.Join(tmpFolder, "image.iso")
+
+		generateO := DefaultGenerateOptions()
+		generateO.ImageType = "iso"
+		generateO.OutputImageFilePath = o.ImageFile
+
+		if err := generateO.Run(ctx, args); err != nil {
+			return err
+		}
+	}
+
+	// generate persistence image
+	if f, err := os.Open(o.PersistenceImagePath); err != nil {
+		zap.S().Infof("failed to find persistence image %q. Generating a new one...", err)
+		output, err := generatePersistentQCOW2(o.PersistenceImagePath)
+		if err != nil {
+			return err
+		}
+		zap.S().Infof("persistent image wrote:%s", output)
+	} else {
+		f.Close()
+	}
+
+	// create domain defintion
+	tmpl, err := template.New("domainDefinition").Parse(libvirtDomainDefinitionTemplate)
+	if err != nil {
+		return err
+	}
+
+	var domainDefBuilder strings.Builder
+	if err := tmpl.Execute(&domainDefBuilder, domainParameters{
+		PersistenceFilePath: o.PersistenceImagePath,
+		ImagePath:           o.ImageFile,
+		Network:             o.NetworkName,
+		Name:                o.Name,
+	}); err != nil {
+		return fmt.Errorf("failed to generate libvirt domain definition: %s", err)
+	}
+
+	conn, err := libvirt.NewConnect(o.QemuUrl)
+	if err != nil {
+		return fmt.Errorf("failed to connect to hypervisor: %v", err)
+	}
+
+	domain, err := conn.DomainDefineXML(domainDefBuilder.String())
+	if err != nil {
+		return fmt.Errorf("failed to define domain: %v", err)
+	}
+	defer func() {
+		_ = domain.Free()
+	}()
+
+	// Start the domain
+	if err := domain.Create(); err != nil {
+		return fmt.Errorf("failed to create domain: %v", err)
+	}
+
+	fmt.Printf("agent started: %s", o.Name)
+
+	return nil
+}
+
+func generatePersistentQCOW2(path string) (string, error) {
+	fmt.Println(path)
+	sshCmd := exec.Command("qemu-img", "create", "-f", "qcow2", path, "1G")
```

Comment:
Can we be consistent and use here libvirt SDK?

Reply:
> Can we be consistent and use here libvirt SDK?

I see there is no support it seems...

Reply:
So can you please at least pass the `--system parameter`?

Reply:
> So can you please at least pass the `--system` parameter (qemu-url)

I see there is no as well :/ so that could cause issue when different `qemu-url` is used. .. anyway, let' procceed